sdk/core/azure-core-amqp/inc/azure/core/amqp/models/amqp_value.hpp

```diff
@@ -70,146 +76,668 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models {
+
+    /** @brief Interoperability helper function which converts an AmqpValue to a uAMQP
+     * AMQP_VALUE object.
+     *
+     * @returns uAMQP AMQP_VALUE object.
+     *
+     * @remarks This is an internal operator which should not be called by customers.
+     */
     operator AMQP_VALUE_DATA_TAG*() const;
+
+    /** @brief Interoperability helper function which creates an AmqpValue from a uAMQP
+     * AMQP_VALUE object.
+     *
+     * @param value source uAMQP AMQP_VALUE object.
+     *
+     * @remarks This is an internal operator which should not be called by customers.
+     */
     AmqpValue(AMQP_VALUE_DATA_TAG* value);
 
+    /** @brief Copy an AMQP value to the current AMQP value.
+     *
+     * @param that the other AMQP Value to copy.
+     * @returns "this".
+     */
     AmqpValue& operator=(AmqpValue const& that);
-    AmqpValue& operator=(AmqpValue&& that) throw();
+    /** @brief Move an AMQP value to the current AMQP value.
+     *
+     * @param that the other AMQP Value to move.
+     * @returns "this".
+     */
+    AmqpValue& operator=(AmqpValue&& that) noexcept;
+
+    /** @brief Equality comparison operator.
+     * @param that - Value to compare to this value.
+     * @returns true if the that is equal to this.
+     */
+    bool operator==(AmqpValue const& that) const;
+    /** @brief Less Than comparison operator.
+     * @param that - Value to compare to this value.
+     * @returns true if the that is less than this.
+     */
+    bool operator<(AmqpValue const& that) const;
+    AmqpValueType GetType() const;
 
+    /** @brief Returns 'true' if the AMQP value is "null".
+     *
+     * @returns true if the AmqpValue is null.
+     */
     bool IsNull() const;
 
+    /** @brief convert the current AMQP Value to a boolean.
+     *
+     * @returns bool true if the AMQP value is true.
+     *
+     * @throws std::runtime_error if the underlying AMQP value is not a boolean.
+     */
     operator bool() const;
-    operator bool();
-    operator unsigned char() const;
-    operator unsigned char();
-    operator char() const;
-    operator char();
-    operator uint16_t() const;
-    operator uint16_t();
-    operator int16_t() const;
-    operator int16_t();
-    operator uint32_t() const;
-    operator uint32_t();
-    operator int32_t() const;
-    operator int32_t();
-    operator uint64_t() const;
-    operator uint64_t();
-    operator int64_t() const;
-    operator int64_t();
+    operator std::uint8_t() const;
+    operator std::int8_t() const;
+    operator std::uint16_t() const;
+    operator std::int16_t() const;
+    operator std::uint32_t() const;
+    operator std::int32_t() const;
+    operator std::uint64_t() const;
+    operator std::int64_t() const;
     operator float() const;
-    operator float();
     operator double() const;
-    operator double();
-    operator BinaryData() const;
-    operator BinaryData();
     explicit operator std::string() const;
```

Comment:
I made it explicit because I found several bugs because of implicit conversions to std::string. To remove those bugs, I made the std::string conversions explicit.